worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,459 @@
+
+        # if one fragment per roll, just add "Roll" objects
+        if frags_per_roll == 1:
+            self.itempool += ["Roll"] * (num_of_rolls - 1)  # minus one because one is in start inventory
+        else:
+            self.itempool += ["Roll"]  # always add a full roll to make generation easier (will be early)
+            self.itempool += ["Roll Fragment"] * (frags_per_roll * (num_of_rolls - 2))
+
+        already_items = len(self.itempool) + self.extra_plando_items
+
+        # Yacht Dice needs extra filler items so it doesn't get stuck in generation.
+        # For now, we calculate the number of extra items we'll need later.
+        if self.options.minimize_extra_items.value:
+            extra_percentage = max(0.1, 0.8 - self.multiworld.players / 10)
+        else:
+            extra_percentage = 0.7
+        extra_locations_needed = max(10, math.ceil(already_items * extra_percentage))
+
+        # max score is the value of the last check. Goal score is the score needed to 'finish' the game
+        self.max_score = self.options.score_for_last_check.value
+        self.goal_score = min(self.max_score, self.options.score_for_goal.value)
+
+        # Yacht Dice adds items into the pool until a score of at least 1000 is reached.
+        # the yaml contains weights, which determine how likely it is that specific items get added.
+        # If all weights are 0, some of them will be made to be non-zero later.
+        weights = [
+            self.options.weight_of_dice.value,
+            self.options.weight_of_roll.value,
+            self.options.weight_of_fixed_score_multiplier.value,
+            self.options.weight_of_step_score_multiplier.value,
+            self.options.weight_of_double_category.value,
+            self.options.weight_of_points.value,
+        ]
+
+        # if the player wants extra rolls or dice, fill the pool with fragments until close to an extra roll/dice
+        if weights[0] > 0 and frags_per_dice > 1:
+            self.itempool += ["Dice Fragment"] * (frags_per_dice - 1)
+        if weights[1] > 0 and frags_per_roll > 1:
+            self.itempool += ["Roll Fragment"] * (frags_per_roll - 1)
+
+        # calibrate the weights, since the impact of each of the items is different
+        weights[0] = weights[0] / 5 * frags_per_dice
+        weights[1] = weights[1] / 5 * frags_per_roll
+
+        extra_points_added = 0
+        multipliers_added = 0
+        items_added = 0
+
+        def get_item_to_add():
+            nonlocal weights
+            nonlocal extra_points_added
+            nonlocal multipliers_added
+            nonlocal items_added
+
+            items_added += 1
+
+            all_items = self.itempool + self.precollected
+            dice_fragments_in_pool = all_items.count("Dice") * frags_per_dice + all_items.count("Dice Fragment")
+            if dice_fragments_in_pool + 1 >= 9 * frags_per_dice:
+                weights[0] = 0  # don't allow >=9 dice
```

Comment:
Hmmm

I wonder if this code would be clearer if `weights` was a `Dict[str, float]`
And you modified it like `weights["Dice"] = 0`

And then at the end, you do `which_item_to_add = self.random.choice(weights.keys(), weights=weights.values)`

And then after that, you can go ahead with `if which_item_to_add == "Dice"`

You feel me?

I think that would be a big improvement actually so I'm not gonna mark this Low Priority, I'd rather you do it or comment on why you don't want to